code/modules/antagonists/bloodsuckers/bloodsucker_objectives.dm

```diff
@@ -93,15 +93,24 @@
 
 // GENERATE!
 /datum/objective/bloodsucker/protege/New()
+
+	var/list/departments_to_list = list(
+		"Security" = GLOB.security_positions,
+		"Engineering" = GLOB.engineering_positions,
+		"Medical" = GLOB.medical_positions,
+		"Science" = GLOB.science_positions,
+		"Supply" = GLOB.supply_positions,
+	)
+
 	switch(rand(0, 2))
 		// Vasssalize Command/QM
 		if(0)
 			target_amount = 1
 			target_department = VASSALIZE_COMMAND
 		// Vassalize a certain department
 		else
-			target_amount = rand(2,3)
 			target_department = pick(departments)
+			target_amount = clamp(rand(2,3), 0, LAZYLEN(departments_to_list[target_department]))
```

Comment:
departments_to_list only contains references to lists of what jobs there are, not how many people are actually playing in that department

Reply:
going to harvest the check completion code